### DISCUSSION TITLE
Issue Running Morphologist Pipeline with Docker BrainVISA

### DISCUSSION THREAD

**Discussion by caro0031** [brainvisa/brainvisa.github.io#171] [Q&A] [answered]
Hello,

I'm experiencing an issue when running the Morphologist pipeline using my BrainVISA Docker that I don't have with my local version of BrainVISA. (The docker is made with the casa_distro_admin command.)

I wrote the following file script in /tmp/tmp_l7tnlv4/temp.py : 
from brainvisa.processes import *
context = defaultContext()
context.runProcess('Morphologist', t1mri='/home/username/brainvisaDataBases/epiSTIM/brainvisaDB/subjects/pat_Name/t1mri/default_acquisition/pat_Name.nii.gz', perform_normalization=False)

The command with the local version is : brainvisa -e /tmp/tmp_l7tnlv4/temp.py -b --noMainWindow 

Here is the command I use to run BrainVISA in Docker (with a mounted volume for the database and one for the script) : 
docker run -it --rm \
  -v /tmp/tmp_l7tnlv4:/temp \
  -v /home/username/brainvisaDataBases/epiSTIM/brainvisaDB:/home/username/brainvisaDataBases/epiSTIM/brainvisaDB \
  brainvisa-5.1.1:5.3 brainvisa -e /temp/temp.py -b --noMainWindow

However, I get the following error:
ERROR: Mandatory argument _t1mri_nobias_ of morphologist_1 has no value
**RuntimeError**
    Traceback (most recent call last):
      File "/casa/install/python/brainvisa/axon/processes.py", line 182, in initializeProcesses
        six.exec_(code, localsStartup, localsStartup)
      File "<neuroConfig.startup>", line 3, in <module>
      File "/temp/temp.py", line 3, in <module>
        context.runProcess('Morphologist',t1mri='/home/username/brainvisaDataBases/epiSTIM/brainvisaDB/subjects/pat_Name/t1mri/default_acquisition/pat_Name.nii.gz',perform_normalization=False)
      File "/casa/install/python/brainvisa/processes.py", line 3112, in runProcess
        self._setArguments(_process, *args, **kwargs)
      File "/casa/install/python/brainvisa/processes.py", line 3073, in _setArguments
        _process.checkArguments()
      File "/casa/install/python/brainvisa/processes.py", line 957, in checkArguments
        o.checkValue(p, getattr(self, p, None))
      File "/casa/install/python/brainvisa/data/writediskitem.py", line 71, in checkValue
        Parameter.checkValue(self, name, value)
      File "/casa/install/python/brainvisa/data/neuroData.py", line 220, in checkValue
        raise RuntimeError(
    RuntimeError: Mandatory argument <em>t1mri_nobias</em> of morphologist_1 has no value

It seems that I need to explicitly specify the arguments for the Morphologist process, which is not the case when using my local version of BrainVISA. I would like to be able to run the script in the same way as on my local version.

Do you have any suggestions or solutions to resolve this issue?

Thank you in advance for your help!
Best,
Carole-Anne

**Comment by denisri** [comment, marked as answer]
Hi,
It seems that the database completion is not done in the command. This suggests that the database is not configured or not found in the docker container. Can you run brainvisa interactively (with graphical interface) once using the docker version, and check that the database settings are OK ? Perhaps also the mount points in the docker container are different from the ones in your "local" version (apptainer?) ?
Denis

**Reply by caro0031** [reply]
Hi,

Thank you for your reply that really helped me.
Indeed, the problem came from the fact that I didn't configure the database in the docker.
By executing the brainvisa configuration GUI in a shell file before running my pipeline, I was able to get my pipeline working after configuring it.

However, I'd like to know if it's possible to do the configuration without going through the GUI, but simply with a command in a batch mode that would do the configuration directly without needing user intervention via the interface ?

Thanks in advance,

Carole-Anne

**Comment by denisri** [comment]
Yes you can use the process "createdatabase":
```
brainvisa -r createdatabase /home/casimir/nice_data
```
Denis

**Reply by caro0031** [reply]
Perfect, it works.
Thank you very much !

Carole-Anne
